### ISSUE TITLE
Can't register an account if captcha/registration form filled incorrectly/slowly.

### ISSUE THREAD

**Issue by gerg5c42g542g2c54g52c** [siacs/Conversations#3452]
#### General information

* **Version:** any
* **Device:** any
* **Android Version:** any
* **Server name:** creep.im, probably any with captcha
* **Server software:** ejabberd
* **Installed server modules:** full compatibility
* **Conversations source:** Conversations Legacy PlayStore and 2y old Conversations F-Droid

#### Steps to reproduce

Register an account on a server that uses captcha via Conversations. The problem occurs in one of the following cases (not sure, but it should happen after a few tries):
if captcha is filled incorrectly
if captcha isn't filled, but the back button is hit instead and then another captcha entry attempt.
if the registration form isn't filled promptly - not quickly, but it takes a while for whatever reason (can't remember exact conditions)

#### Expected result

Account should be registered (I made sure that the username wasn't taken) after entering captcha, even if it wasn't entered correctly before.

#### Actual result

An error. Error is shown every time, even if going back to Conversations' home screen and doing everything again from scratch, even if using another username (also not taken), if using another password, if entering captcha correctly.
An app reinstall magically fixes the problem.

#### Debug output

Not my phone, no debug. It happened when a newly met person tried using Conversations, because I don't use Facebook. They couldn't create the account and even when they gave me the phone to try, I couldn't do it either no matter what I did. This resulted in everyone becoming really confused. Reinstall helped. I'm sure it wasn't a server issue because another newly met person did that successfully at the same moment, albeit they didn't do any typos and filled everything fast. I remember encountering this same bug some 2 years ago when I was making my own XMPP account or some test accounts.

**Comment by gerg5c42g542g2c54g52c** [comment]
I confirmed this happens in the latest Conversations 2.5.1 too.

The error is triggered consistently by the following. Almost always if not always. I'm pretty sure it's triggered by different actions too.

To trigger the bug:
- Open Conversations.
- Tap Create Account (on conversations.im)
- Enter username
- Hit Next
- When captcha appears, tap Cancel and go back to the main screen
- Tap Use My Own Provider
- Fill the form (use creep.im server) and hit Next
- When captcha appears, tap Cancel and hit Next again
- You should see a "Registration failed: Try again later" error after solving captcha.
- Try a few more times, you'll see it's not possible to register an account unless the app is reinstalled.

Debug log:

````
05-09 21:29:48.827 13408 13408 D conversations: initializing database...
05-09 21:29:48.828 13408 13408 D conversations: restoring accounts...
05-09 21:29:48.847 13408 13408 D conversations: restoring conversations...
05-09 21:29:48.848 13408 13408 D conversations: finished restoring conversations in 1ms
05-09 21:29:48.856 13408 13430 D conversations: restoring roster...
05-09 21:29:48.857 13408 13430 D conversations: restoring messages...
05-09 21:29:48.857 13408 13430 D conversations: update unread count to 0
05-09 21:29:48.858 13408 13428 D conversations: start merging phone contacts with roster
05-09 21:29:48.858 13408 13428 D conversations: finished merging phone contacts
05-09 21:29:48.859 13408 13408 D conversations: stopped foreground service
05-09 21:29:48.862 13408 13432 D conversations: updated dynamic shortcuts
05-09 21:29:48.872 13408 13430 D conversations: finished restoring messages in 16ms
05-09 21:29:48.925 13408 13408 D conversations: EventReceiver ignored action android.net.conn.CONNECTIVITY_CHANGE
05-09 21:29:48.925 13408 13408 D conversations: app switched into foreground
05-09 21:29:49.139 13408 13408 D conversations: app switched into background
05-09 21:30:05.934 13408 13408 I conversations: AxolotlService (redacted@conversations.im): Could not retrieve axolotl registration id for account redacted@conversations.im/mobile
05-09 21:30:05.935 13408 13408 I conversations: AxolotlService : Generating axolotl registration ID...
05-09 21:30:05.947 13408 13408 W conversations: AxolotlService (redacted@conversations.im): Could not retrieve current prekey id for account redacted@conversations.im/mobile
05-09 21:30:05.959 13408 13408 D conversations: stopped foreground service
05-09 21:30:05.980 13408 13478 D conversations: redacted@conversations.im: sending offline presence
05-09 21:30:05.982 13408 13478 D conversations: redacted@conversations.im do not write stanza to unbound stream <presence type="unavailable" from="redacted@conversations.im/phone"/>
05-09 21:30:05.983 13408 13478 D conversations: redacted@conversations.im: disconnecting force=false
05-09 21:30:05.983 13408 13478 D conversations: redacted@conversations.im: socket was null during force close
05-09 21:30:06.227 13408 13408 D conversations: app switched into foreground
05-09 21:30:08.091 13408 13482 D conversations: redacted@conversations.im: socket was null during force close
05-09 21:30:08.091 13408 13482 D conversations: redacted@conversations.im: connecting
05-09 21:30:08.093 13408 13408 D conversations: started foreground service
05-09 21:30:13.437 13408 13482 D conversations: Resolver: [Result{ip='redacted', hostame='xmpps.conversations.im', port=443, directTls=true, authenticated=false, priority=1}, Result{ip='redacted', hostame='xmpps.conversations.im', port=443, directTls=true, authenticated=false, priority=1}, Result{ip='redacted', hostame='xmpp.conversations.im', port=5222, directTls=false, authenticated=false, priority=5}, Result{ip='redacted', hostame='xmpp.conversations.im', port=5222, directTls=false, authenticated=false, priority=5}, Result{ip='redacted', hostame='xmpps.conversations.im', port=80, directTls=false, authenticated=false, priority=10}, Result{ip='redacted', hostame='xmpps.conversations.im', port=80, directTls=false, authenticated=false, priority=10}]
05-09 21:30:13.449 13408 13482 D conversations: redacted@conversations.im: using values from dns xmpps.conversations.im/redacted:443 tls: true
05-09 21:30:21.319 13408 13408 D conversations: app switched into background
05-09 21:30:22.001 13408 13408 D conversations: app switched into foreground
05-09 21:30:32.589 13408 13408 I conversations: AxolotlService (redacted@creep.im): Could not retrieve axolotl registration id for account redacted@creep.im/mobile
05-09 21:30:32.589 13408 13408 I conversations: AxolotlService : Generating axolotl registration ID...
05-09 21:30:32.593 13408 13408 W conversations: AxolotlService (redacted@creep.im): Could not retrieve current prekey id for account redacted@creep.im/mobile
05-09 21:30:32.616 13408 13408 D conversations: started foreground service
05-09 21:30:32.633 13408 13535 D conversations: redacted@creep.im: socket was null during force close
05-09 21:30:32.634 13408 13535 D conversations: redacted@creep.im: connecting
05-09 21:30:32.871 13408 13535 D conversations: Resolver: [Result{ip='redacted', hostame='creep.im', port=443, directTls=true, authenticated=false, priority=10}, Result{ip='redacted', hostame='creep.im', port=5222, directTls=false, authenticated=false, priority=10}]
05-09 21:30:32.876 13408 13535 D conversations: redacted@creep.im: using values from dns creep.im/redacted:443 tls: true
05-09 21:30:42.651 13408 13408 D conversations: started foreground service
05-09 21:30:42.660 13408 13554 D conversations: redacted@creep.im: connecting
05-09 21:30:42.701 13408 13554 D conversations: Resolver: [Result{ip='redacted', hostame='creep.im', port=443, directTls=true, authenticated=false, priority=10}, Result{ip='redacted', hostame='creep.im', port=5222, directTls=false, authenticated=false, priority=10}]
05-09 21:30:42.701 13408 13554 D conversations: redacted@creep.im: using values from dns creep.im/redacted:443 tls: true
05-09 21:30:50.532 13408 13408 D conversations: started foreground service
05-09 21:30:50.542 13408 13570 D conversations: redacted@creep.im: connecting
05-09 21:30:50.607 13408 13570 D conversations: Resolver: [Result{ip='redacted', hostame='creep.im', port=443, directTls=true, authenticated=false, priority=10}, Result{ip='redacted', hostame='creep.im', port=5222, directTls=false, authenticated=false, priority=10}]
05-09 21:30:50.610 13408 13570 D conversations: redacted@creep.im: using values from dns creep.im/redacted:443 tls: true
````

**Comment by licaon-kter** [comment]
Following your steps...I've canceled a dozen captchas on creep.im until it failed, yet then I registered a new account ok.

**Comment by gerg5c42g542g2c54g52c** [comment]
@licaon-kter have you followed all my instructions from the second post correctly? Have you tried a few times?

**Comment by iNPUTmice** [comment]
'Registration failed - Try again later' is an error issued by the server.
Usually due to rate limits. You should do exactly this. Try again later.
No need to reinstall the app.

**Comment by licaon-kter** [comment]
@gerg5c42g542g2c54g52c yes...

**Comment by gerg5c42g542g2c54g52c** [comment]
The problem is not why it's happening, but that it is happening. This is horrible UX no matter the cause of it.

See, a newly met friend installed the app. They tried creating an account, but they couldn't, because of this. They didn't do anything a normal person wouldn't do. They didn't abuse the server in any way to be blocked, they aren't a bot, they aren't malicious. They tried a maximum of two times, this isn't - if it's rate limiting that they experienced then that rate limiting is nuts. They simply did the same thing they do when they install Whatsapp or Wire or everything else. But they have experienced something they don't experience with the other apps. Even I couldn't figure it out. And reinstalling helped, so I'm betting it influences this somehow. If I didn't try reinstalling at that time we would just give up and use something else. I experienced this at least two times when using the app normally (not testing or trying to trigger the bug), so this wasn't just some rare 1 in a million sort of a bug.

I think it's clear that this needs to be fixed. I don't know the cause of this, I tried providing as much info as I could. If this is a server-side problem, I can CC it to ejabberd bugtracker.

I also have seen some Session something errors while debugging this, no idea if related to this.
